packages/apollo-server-core/src/types.ts

```diff
@@ -64,6 +65,25 @@ type BaseConfig = Pick<
   | 'cache'
 >;
 
+export type Unsubscriber = () => void;
+export type SchemaChangeCallback = (schema: GraphQLSchema) => void;
+
+export type GraphQLServiceConfig = {
+  schema: GraphQLSchema;
+  executor: GraphQLExecutor;
+};
+
+export type EngineConfig = {
+  apiKeyHash: string;
+  graphId: string;
+  graphTag?: string;
```

Comment:
We're pretty close to rolling out the term "variant" to replace the idea of "tag".  Since this is a new API, and one which we only use internally right now, I wonder if we might just go ahead and use `graphVariant` now?  (Even as it stands currently, nobody outside Apollo will likely understand that `graphId` is the same as a `serviceId`, but I think we should keep `graphId` for the same reason I just mentioned about `graphVariant`.)